File: Java/Exercism/06GigaSecond/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    private final LocalDateTime time;

    public Gigasecond(LocalDate date) {
        this.time = date.atStartOfDay();
    }

    public Gigasecond(LocalDateTime date) {
        this.time = date;
    }

    public LocalDateTime getDateTime() {
        return time.plusSeconds(1_000_000_000L);
    }
}
/*
 * This is a challenge on Exercism.
 * Link ---> https://exercism.org/tracks/java/exercises/gigasecond
 */

File: Kotlin/Exercism/04Gigasecond/Iteration1/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

class Gigasecond(input: LocalDateTime) {
    val date: LocalDateTime = input.plusSeconds(1_000_000_000)

    constructor(input: LocalDate) : this(input.atStartOfDay())
}
/*
* This is a challenge in Exercism.org
* Link --->  https://exercism.org/tracks/kotlin/exercises/gigasecond
* This is my first try - Iteration 1.
* */